### PR TITLE
Mimir VFS Rewrite

### LINKED ISSUE #2251: Make the ManageFile stuff less… insane
It's all super-hacky and dumb. Causes errors. Etc.  It should be not dumb.

### LINKED ISSUE #2320: Rework VFS actor flow in terms of fs2.Stream
Similar to #2165, but just for the VFS stuff.

### PR DESCRIPTION
**Pull request by djspiewak**
Ripped out the things.  Put in the new ones.  Integration may or may not pass.  Will fix.

https://gist.github.com/djspiewak/2dcb767e52ea1106a2776cc730805d2b

Fixes #2320
Fixes #2251

### CODE REVIEW

**Comment by alissapajer** [comment]
Error on travis in a spec that this PR adds:
```
[error]   x list the contents of a directory
[error]    List(FileIn(Current,FileName(test1)), FileIn(Current,FileName(test3)), FileIn(Current,FileName(test2)))
[error]    
[error]     is not equal to 
[error]    
[error]    List(FileIn(Current,FileName(test1)), FileIn(Current,FileName(test2)), FileIn(Current,FileName(test3))) (RealPOSIXSpecs.scala:105)
```

**Comment by djspiewak** [comment]
@alissapajer Well that's not weird or anything…

I'll look at that.  I'm also adding a `moveDir` function to the vfs, since I forgot to do that and the integration tests rely on that functionality.

**Comment by alissapajer** [comment]
And then the `mimir` build failed with:
```
unimplemented value of type quasar.mimir.Mimir.Backend[slamdata.Predef.Unit] at Mimir.scala:435
```

**Comment by alissapajer** [comment]
@djspiewak also, can we make all the integration tests for mimir "pending" instead of "skip" in this PR? That way they will all be run on travis.

**Comment by djspiewak** [comment]
@alissapajer Agreed, though I have no idea how to do that. Will you be around to pair a bit on Monday?

**Comment by alissapajer** [comment]
@djspiewak I'm around all of Monday!

Re "skip" to "pending" it's just a matter of a search and replace of all the integration test json. I'd be happy to PR that to this branch.

File: mimir/src/main/scala/quasar/mimir/Precog.scala [hunk 1]
```diff
@@ -47,34 +42,50 @@ import akka.routing.{
...
-    val storageTimeout: FiniteDuration = new FiniteDuration(300, SECONDS)
     val quiescenceTimeout: FiniteDuration = new FiniteDuration(300, SECONDS)
     val maxOpenPaths: Int = 500
     val dataDir: File = dataDir0
   }
 
+  val CookThreshold: Int = 20000
+  val StorageTimeout: FiniteDuration = 300.seconds
+
+  private var _vfs: SerialVFS = _
```
Comment (alissapajer) [inline mimir/src/main/scala/quasar/mimir/Precog.scala]: What is this?

**Pull request merged**